### PR TITLE
Allow multiple image selections when adding favicons

### PR DESCRIPTION
**Pull request by droidmonkey**
## Description

Quick tweak to let users select multiple icons when adding favicons from the filesystem. Useful for testing too!

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/gui/EditWidgetIcons.cpp [hunk 1]
```diff
@@ -314,14 +314,15 @@ void EditWidgetIcons::addCustomIconFromFile()
...
-            } else {
-                emit messageEditEntry(tr("Can't read icon"), MessageWidget::Error);
+        auto filenames = QFileDialog::getOpenFileNames(this, tr("Select Image(s)"), "", filter);
+        for (const auto& filename : filenames) {
+            if (!filename.isEmpty()) {
+                auto icon = QImage(filename);
+                if (!icon.isNull()) {
+                    addCustomIcon(QImage(filename));
+                } else {
+                    emit messageEditEntry(tr("Can't read icon"), MessageWidget::Error);
```
Comment (TheZ3ro) [inline src/gui/EditWidgetIcons.cpp]: An improvement: report in the message the filename that can't be loaded.

Also if I load 100 corrupted icons this message will be "spammed" 100 times.
IMHO this needs to be reported at the end of the for cycle with a list of the error filenames.
If the list is too long display only the first `N` filename

Comment (droidmonkey) [reply src/gui/EditWidgetIcons.cpp]: This was way more complicated than it needed to be, but after some replumbing it woks great!

File: src/gui/EditWidgetIcons.cpp [hunk 2]
```diff
@@ -314,43 +317,76 @@ void EditWidgetIcons::addCustomIconFromFile()
...
-                addCustomIcon(QImage(filename));
+        auto filenames = QFileDialog::getOpenFileNames(this, tr("Select Image(s)"), "", filter);
+        if (!filenames.empty()) {
+            QStringList errornames;
+            int numexisting = 0;
+            for (const auto& filename : filenames) {
+                if (!filename.isEmpty()) {
+                    auto icon = QImage(filename);
+                    if (!icon.isNull()) {
+                        if (!addCustomIcon(QImage(filename))) {
```
Comment (phoerious) [inline src/gui/EditWidgetIcons.cpp]: Combine conditions.

File: src/gui/EditWidgetIcons.cpp [hunk 3]
```diff
@@ -314,43 +317,76 @@ void EditWidgetIcons::addCustomIconFromFile()
...
+            QString msg;
+
+            if (numloaded > 0) {
+                msg = tr("Successfully loaded %1 of %2 icons").arg(numloaded).arg(filenames.size());
+            } else {
+                msg = tr("No icons were loaded");
+            }
+
+            if (numexisting > 0) {
+                msg += ", " + tr("%1 icons already existed").arg(numexisting);
```
Comment (phoerious) [inline src/gui/EditWidgetIcons.cpp]: This doesn't work. You cannot assume other languages to append clauses in this order or use commas to separate them. This needs to be two separate messages.

File: src/gui/EditWidgetIcons.cpp [hunk 4]
```diff
@@ -314,43 +317,76 @@ void EditWidgetIcons::addCustomIconFromFile()
...
+                        errornames << filename;
+                    }
+                }
+            }
+
+            int numloaded = filenames.size() - errornames.size() - numexisting;
+            QString msg;
+
+            if (numloaded > 0) {
+                msg = tr("Successfully loaded %1 of %2 icons").arg(numloaded).arg(filenames.size());
```
Comment (phoerious) [inline src/gui/EditWidgetIcons.cpp]: Missing plural handling.

File: src/gui/EditWidgetIcons.cpp [hunk 5]
```diff
@@ -314,43 +317,76 @@ void EditWidgetIcons::addCustomIconFromFile()
...
+            }
+
+            if (numexisting > 0) {
+                msg += ", " + tr("%1 icons already existed").arg(numexisting);
+            }
+
+            if (!errornames.empty()) {
+                // Show the first 8 icons that failed to load
+                errornames = errornames.mid(0, 8);
+                emit messageEditEntry(msg + "\n" + tr("The following icons failed:") + "\n" + errornames.join("\n"),
```
Comment (phoerious) [inline src/gui/EditWidgetIcons.cpp]: See above.

File: src/gui/EditWidgetIcons.cpp (hunk 2 above)
Comment (droidmonkey) [reply src/gui/EditWidgetIcons.cpp]: They have to be separate because I want to fall through (ie, not an error).

Comment (phoerious) [reply src/gui/EditWidgetIcons.cpp]: That would be `else if (icon.isNull())`.

**Comment by droidmonkey** [comment]
Made corrections and rebased onto develop due to complexity.

**Pull request merged**